go/obscuronode/host/host.go

```diff
@@ -400,7 +400,10 @@ func (a *Node) handleRoundWinner(result nodecommon.BlockSubmissionResponse) func
 				Rollup: nodecommon.EncodeRollup(winnerRollup.ToRollup()),
 			}
 
-			a.broadcastTx(a.mgmtContractLib.CreateRollup(tx, a.ethWallet.GetNonceAndIncrement()))
+			if !a.DB().WasSubmitted(winnerRollup.Header.Hash()) {
```

Comment:
Don't submit the rollup if the current header is not submitted?
Is that a correct reading of this change ?

Not sure I follow the underlying logic, can you explain please ?

Reply:
there is a race where that handler gets called twice for the same height. And it will return the same winner from the enclave which will get submitted.
This checks if it was submitted already

Reply:
Cool, can that be captured in the comments for future ref please ?